### DISCUSSION TITLE
Discrepencies in the simSurvey values

### DISCUSSION THREAD

**Discussion by cmmckeon** [NOAA-EDAB/ms-keyrun#86] [Q&A] [answered]
I've been trying to get the values needed to calculate growth curves for mizer.

Something strange is happening with Green_halibut.

I can get the age, length and weight at maturity values from the simSurvey objects. However, if I use the WLa and WLb values from the simBiolPar object and the given length at maturity to calculate the weight at maturity, I get a completely different value to the maturity weight in the simSurveyWtatAge object. 

This is giving me strange growth curves in the mizer model, which do not match the simulated data. Is it possible that the WLa or WLb values are wrong?

Here is the code I am using;

## mean maturity_weight from the given data
w_at_age_5 <- mean(simSurveyWtatAge$value[simSurveyWtatAge$Name == "Green_halibut" & simSurveyWtatAge$age == 5]) ## 3201.84 grams

## weighted mean maturity_length from the given data
l_at_age_5 <- weighted.mean(simSurveyAgeLencomp$lenbin[simSurveyAgeLencomp$Name == "Green_halibut" & simSurveyAgeLencomp$agecl == 5],w=simSurveyAgeLencomp$value[simSurveyAgeLencomp$Name == "Green_halibut" & simSurveyAgeLencomp$agecl == 5]) ##149.28 cm

## given WLa and WLb factors
atlan_a <- simBiolPar$WLa[simBiolPar$Name == "Green_halibut"]
atlan_b <- simBiolPar$WLb[simBiolPar$Name == "Green_halibut"]

## calculate maturity weight from length weight relationship
atlan_a * l_at_age_5 ^ atlan_b ## = 6958.297 grams...

**Comment by cmmckeon** [comment]
Here is the growth curve (plotting on top of the simSurvey data) I get when I use the give WLa and WLb values to calculate weight at maturity

<img width="451" alt="image" src="https://github.com/NOAA-EDAB/ms-keyrun/assets/44772138/19ce76fe-d133-40a5-a313-f944ab2f2b19">

**Comment by cmmckeon** [comment]
Here is the curve I get with a different WLa value (I just used the value for Cod to check what would happen).

![adjusted_WLa_value](https://github.com/NOAA-EDAB/ms-keyrun/assets/44772138/4ba26717-d63e-4252-87a1-2a6914cc5a26)

**Comment by vbartolino** [comment]
@cmmckeon note that ```simSurveyAgeLencomp``` uses age classes and not true ages. Could be that the reason for such discrepancy? Anyway, this would be an issue for all the species (except BWH, CAP, PCO). See ```$AgeperAgecl``` in ```simBiolPar``` to get how many ages are represented in each age class for each species.

**Comment by vbartolino** [comment, marked as answer]
in practice with ```simSurveyAgeLencomp$agecl == 5``` you're looking at fish in the agecl 5 which should refer to fish age ~10
This seems confirmed
```
> mean(simSurveyWtatAge$value[simSurveyWtatAge$Name == "Green_halibut" & simSurveyWtatAge$age == 10])
[1] 6814.51
```
which is not far from 6958.297 grams that you calculated above.
I hope this helps

**Reply by cmmckeon** [reply]
Cheers Valerio - that sorts it out for me
